### PR TITLE
Drop Waf build system support

### PR DESCRIPTION
**Pull request by eht16**
See #768.

### CODE REVIEW

File: HACKING [hunk 1]
```diff
@@ -171,9 +171,9 @@ Geany requires GTK >= 2.24 and GLib >= 2.28. API symbols from newer
 GTK/GLib versions should be avoided or made optional to keep the source
 code building on older systems.
 
-It is recommended to use the 2.24 API documentation of the GTK 
-libs (including GLib, GDK and Pango) has the advantages 
-that you don't get confused by any newer API additions and you 
+It is recommended to use the 2.24 API documentation of the GTK
+libs (including GLib, GDK and Pango) has the advantages
+that you don't get confused by any newer API additions and you
```
Comment (sardemff7) [inline HACKING]: While it’s good to remove trailing white spaces, it should probably not be in this commit.

Comment (eht16) [inline HACKING]: I agree in general. In this case it is probably just not worth to make an extra commit for this.

File: geany.spec.in [hunk 2]
```diff
@@ -60,13 +60,6 @@ Requires:  pkgconfig gtk2-devel
 This package contains the header files and pkg-config file needed for building
 Geany plug-ins. You do not need to install this package to use Geany.
 
-%prep
-%setup -q
```
Comment (b4n) [inline geany.spec.in]: should these 2 lines really be removed?

Comment (eht16) [inline geany.spec.in]: Oops, no. I re-added the missing lines.
Will squash the commits before merging.

**Pull request merged**